ml-prims/src/random/rng.h

```diff
@@ -283,6 +283,26 @@ class Rng {
              NumThreads, nBlocks, type, stream);
   }
 
+  /**
+   * @brief Generate bernoulli distributed array and applies scale
+   * @tparam Type data type in which to compute the probabilities
+   * @tparam LenType data type used to represent length of the arrays
+   * @param ptr the output array
+   * @param len the number of elements in the output
+   * @param prob coin-toss probability for heads
+   * @param stream stream where to launch the kernel
+   */
+  template <typename Type, typename LenType = int>
+  void scaled_bernoulli(Type *ptr, LenType len, Type prob, Type scale,
```

Comment:
We should add a  quick test for this- namely just to make sure results are within the expected range given the inputs.